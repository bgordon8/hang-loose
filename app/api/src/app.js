import express from 'express'
import { json, urlencoded } from 'body-parser'
import users from './routes/users'
import workspaces from './routes/workspaces'

const app = express()

app.use(json())
app.use(urlencoded({ extended: false }))
app.use(users)
app.use(workspaces)

if (process.env.NODE_ENV !== 'test') {
  app.listen(3000, () => {
    console.log('listening on port 3000...')
  })
}

export default app
