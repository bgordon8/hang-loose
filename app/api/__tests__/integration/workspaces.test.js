import supertest from 'supertest'
import app from '../../src/app'

const req = supertest(app)

describe('routes : workspaces', () => {
  describe('GET /workspaces', () => {
    it('returns all workspaces', async () => {
      const res = await req.get('/workspaces')

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('workspaces')
      expect(res.body.workspaces[0]).toHaveProperty('id')
      expect(res.body.workspaces[0].id).toBe(1)
      expect(res.body.workspaces[0]).toHaveProperty('name')
      expect(res.body.workspaces[0].name).toBe('good-guys')
      expect(res.body.workspaces[0]).toHaveProperty('cname')
      expect(res.body.workspaces[0].cname).toBe('good-guys')
      expect(res.body.workspaces[0]).toHaveProperty('ownerId')
      expect(res.body.workspaces[0].ownerId).toBe(1)
      expect(res.body).toMatchSnapshot()
    })
  })
})
