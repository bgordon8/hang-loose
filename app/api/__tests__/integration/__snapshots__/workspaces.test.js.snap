// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes : workspaces GET /workspaces returns all workspaces 1`] = `
Object {
  "status": "success",
  "workspaces": Array [
    Object {
      "cname": "good-guys",
      "created_at": "2020-10-21T06:00:00.000Z",
      "id": 1,
      "name": "good-guys",
      "ownerId": 1,
      "updated_at": "2020-10-21T06:00:00.000Z",
    },
    Object {
      "cname": "misguided-soles",
      "created_at": "2020-10-21T06:00:00.000Z",
      "id": 2,
      "name": "misguided-soles",
      "ownerId": 1,
      "updated_at": "2020-10-21T06:00:00.000Z",
    },
  ],
}
`;
